core/factions/baronyOfLetnev.ts

```diff
@@ -176,4 +176,25 @@ export const baronyOfLetnev: BattleEffect[] = [
     },
     timesPerFight: 1,
   },
+
+  //TODO: Needs better logic about switching the bonus to worse hit ships if the +X causes the highestHitUnit to go over guaranteed hits
+  {
+    name: 'Gravleash Maneuvers',
+    description:
+      "Barony Breakthrough: Before you roll dice during space combat, apply +X to the results of 1 of your ship's rolls, where X is the number of ship types you have in the combat.",
+    type: 'faction-ability',
+    place: Place.space,
+    faction: Faction.barony_of_letnev,
+    onStart: (p: ParticipantInstance, _b: BattleInstance, _op: ParticipantInstance) => {
+      const highestHitUnit = getHighestHitUnit(p, 'combat', Place.space)
+      if (highestHitUnit && highestHitUnit.combat) {
+        const units = getUnits(p, Place.space, true)
+        const numUniqueUnits = [...new Set(units)].length
```

Comment:
This number will be way too high in many situations. It shouldn't count unique units, it should count unique values of the `UnitType` property, excluding `UnitType.other`.